diff --git a/target/riscv/insn32.decode b/target/riscv/insn32.decode
--- a/target/riscv/insn32.decode
+++ b/target/riscv/insn32.decode
@@ -109,6 +109,7 @@ mvm_move     0000101 ..... ..... 111 ..... 0001011 @r
 mvm_vset     0000110 ..... ..... 111 ..... 0001011 @r
 mvm_vl       0000111 ..... ..... 111 ..... 0001011 @r
 mvm_vs       0001000 ..... ..... 111 ..... 0001011 @r
+mvm_configure 0001001 ..... ..... 111 ..... 0001011 @r
 
 # Formats 64:
 @sh5     .......  ..... .....  ... ..... ....... &shift  shamt=%sh5      %rs1 %rd
