diff --git a/Makefile.in b/Makefile.in
index 71870a4..ba35dbc 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1076,6 +1076,7 @@ stamps/build-gcc-musl-stage2: $(GCC_SRCDIR) $(GCC_SRC_GIT) stamps/build-musl-lin
 		$(WITH_ISA_SPEC) \\
 		$(GCC_WITH_SPECS) \\
 		$(GCC_EXTRA_CONFIGURE_FLAGS) \\
+		$(GCC_STAGE2_EXTRA_CONFIGURE_FLAGS) \\
 		CFLAGS_FOR_TARGET="-O2 $(CFLAGS_FOR_TARGET)" \\
 		CXXFLAGS_FOR_TARGET="-O2 $(CXXFLAGS_FOR_TARGET)"
 	$(MAKE) -C $(notdir $@)
